src/nanoarrow/array.c

```diff
@@ -846,6 +856,22 @@ static int ArrowArrayViewValidateMinimal(struct ArrowArrayView* array_view,
         return EINVAL;
       }
       break;
+
+    case NANOARROW_TYPE_RUN_END_ENCODED:
+      if (array_view->children[0]->null_count != 0) {
+        ArrowErrorSet(error, "Run Ends cannot be null but found null_count value %ld",
+                      (long)array_view->children[0]->null_count);
+        return EINVAL;
+      }
+      if (array_view->children[0]->length < array_view->children[1]->length) {
+        ArrowErrorSet(
+            error,
+            "Expected the 2 children of run-end encoded array to have equal length "
+            "but found mismatched lengths: run_ends->length=%ld, values->length=%ld",
+            (long)array_view->children[0]->length, (long)array_view->children[1]->length);
+        return EINVAL;
+      }
```

Comment:
It's OK to have less run-ends than values. It's the opposite that's wrong because you always start the indexing of values by checking the run-ends first:

1. find run-end index according to logical index
2. use that index with the values array

https://github.com/apache/arrow/blob/main/cpp/src/arrow/util/ree_util.cc#L199